sync/class.jetpack-sync-module.php

```diff
@@ -68,35 +68,47 @@ protected function enqueue_all_ids_as_action( $action_name, $table_name, $id_fie
 		$items_per_page  = 1000;
 		$page            = 1;
 		$chunk_count     = 0;
-		$previous_max_id = $state ? $state : '~0';
+		$previous_interval_endpoint = $state ? $state : '~0';
 		$listener        = Jetpack_Sync_Listener::get_instance();
 
 		// count down from max_id to min_id so we get newest posts/comments/etc first
-		while ( $ids = $wpdb->get_col( "SELECT {$id_field} FROM {$table_name} WHERE {$where_sql} AND {$id_field} < {$previous_max_id} ORDER BY {$id_field} DESC LIMIT {$items_per_page}" ) ) {
+		while ( $ids = $wpdb->get_col( "SELECT {$id_field} FROM {$table_name} WHERE {$where_sql} AND {$id_field} < {$previous_interval_endpoint} ORDER BY {$id_field} DESC LIMIT {$items_per_page}" ) ) {
 			// Request posts in groups of N for efficiency
 			$chunked_ids = array_chunk( $ids, self::ARRAY_CHUNK_SIZE );
 
 			// if we hit our row limit, process and return
 			if ( $chunk_count + count( $chunked_ids ) >= $max_items_to_enqueue ) {
 				$remaining_items_count = $max_items_to_enqueue - $chunk_count;
 				$remaining_items       = array_slice( $chunked_ids, 0, $remaining_items_count );
-
-				$listener->bulk_enqueue_full_sync_actions( $action_name, $remaining_items );
+				$remaining_items_with_previous_interval_endpoints = $this->get_chunks_with_preceding_endpoint( $remaining_items, $previous_interval_endpoint );
+				$listener->bulk_enqueue_full_sync_actions( $action_name, $remaining_items_with_previous_interval_endpoints );
 
 				$last_chunk = end( $remaining_items );
 				return array( $remaining_items_count + $chunk_count, end( $last_chunk ) );
 			}
+			$chunked_ids_with_previous_interval_endpoints = $this->get_chunks_with_preceding_endpoint( $chunked_ids, $previous_interval_endpoint );
 
-			$listener->bulk_enqueue_full_sync_actions( $action_name, $chunked_ids );
+			$listener->bulk_enqueue_full_sync_actions( $action_name, $chunked_ids_with_previous_interval_endpoints );
 
 			$chunk_count    += count( $chunked_ids );
 			$page           += 1;
-			$previous_max_id = end( $ids );
+			$previous_interval_endpoint = end( $ids );
```

Comment:
Because you use `min` in `get_chunks_with_preceding_endpoint` I'm assuming that id lists are not sorted. Should we instead use `max` here then?

Reply:
Edit: I can see that the array is actually sorted by IDs, makes sense to use `end` here then. But what about `min` in the method down there? Can you comment somewhere in the code about that so that it's more obvious why these choices are made? Thanks!

Reply:
added a comment and now I am always using end()